extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletHttpSecurityPolicy.java

```diff
@@ -42,21 +44,31 @@
             } else if (emptyRoleSemantic == SecurityInfo.EmptyRoleSemantic.DENY) {
                 return Uni.createFrom().item(CheckResult.DENY);
             } else if (emptyRoleSemantic == SecurityInfo.EmptyRoleSemantic.AUTHENTICATE) {
-                if (identity.isAnonymous()) {
-                    return Uni.createFrom().item(CheckResult.DENY);
-                } else {
-                    return Uni.createFrom().item(CheckResult.PERMIT);
-                }
+                return identity.map(new Function<SecurityIdentity, CheckResult>() {
+                    @Override
+                    public CheckResult apply(SecurityIdentity securityIdentity) {
+                        if (securityIdentity.isAnonymous()) {
```

Comment:
So if the proactive authentication is disabled then the access will be denied if the resource is secured ? I think this is what you said earlier would happen in such a case, which makes sense